packages/core/src/agent/index.ts

```diff
@@ -701,17 +703,25 @@ export class Agent<
       output?: OutputType | Z;
     } = {},
   ): Promise<StreamReturn<Z>> {
-    let messagesToUse: UserContent[] = [];
+    let messagesToUse: CoreMessage[] = [];
 
-    if (Array.isArray(messages)) {
-      messagesToUse = messages.map(content => {
-        if (typeof content === 'string') {
-          return content;
+    if (typeof messages === `string`) {
+      messagesToUse = [
+        {
+          role: 'user',
+          content: messages,
+        },
+      ];
+    } else {
+      messagesToUse = messages.map(message => {
+        if (typeof message === `string`) {
+          return {
+            role: 'user',
+            content: message,
+          };
         }
-        return content.content as string;
+        return message;
```

Comment:
Here we were taking any structured message and discarding everything except content. If the role was `assistant`, `system`, or `tool` it would be converted to `user`